File: 1133-LargestUniqueNumber/1133-LargestUniqueNumber.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var largestUniqueNumber = function(nums) {
    visited = new Map()
    ans = -Infinity
    nums.forEach((num, i) => {
        visited.set(num, (visited.get(num) || 0) + 1);
    })
    for(let [key, frequency] of visited){
        console.log(frequency)
        if(frequency < 2){
            ans = Math.max(ans, key);
        }
    }

    return  nums.length == 0 || ans == -Infinity ? -1 : ans
};

File: 1133-LargestUniqueNumber/1133-LargestUniqueNumber.py
class Solution:
    def largestUniqueNumber(self, nums: List[int]) -> int:
        nums_with_frequencies = {}
        ans = float('-inf')
        for num in nums:
            nums_with_frequencies[num] = nums_with_frequencies.get(num, 0) + 1
        
        for key, val in nums_with_frequencies.items():
            if val < 2:
                ans = max(ans, key)
        
        return -1 if len(nums) == 0 or ans == float('-inf') else ans


        
